### PR TITLE
Improved event update

### PR DESCRIPTION
**Pull request by LaChope**
Do not resolve any specific issue, it it a global improvement including: 
- Root events can no longer be updated (read only)
- External events do not have probability field anymore

### CODE REVIEW

**Comment by blcham** [comment]
> External events do not have probability field anymore


@LaChope can you explain this? They should have probability

File: src/components/dialog/faultEvent/FaultEventCreation.tsx
Comment (blcham) [inline src/components/dialog/faultEvent/FaultEventCreation.tsx:86]: consider renaming variable

eventTypeWatch -- strange name convention, not sure what i should imagine behind that. Can we be more specific?

File: src/components/dialog/faultEvent/FaultEventCreation.tsx
Comment (blcham) [inline src/components/dialog/faultEvent/FaultEventCreation.tsx:190]: gateTypeWatch ? strange name .... see comment above.

File: src/components/dialog/faultEvent/FaultEventCreation.tsx
Comment (blcham) [inline src/components/dialog/faultEvent/FaultEventCreation.tsx:192]: This is a duplicate and can be removed see line 190

**Review by blcham — CHANGES_REQUESTED** [review]
See my comments.

**Comment by blcham** [comment]
> @blcham Ediding an event is still not perfect yet, but I think it makes sense to wait for #281 to be completed first.

I assume that you do not want me to merge until i do not merge #281

File: src/utils/utils.tsx
Comment (LaChope) [reply src/utils/utils.tsx:110]: I am not sure what you mean by that? something like: `const updateEventsType: FaultEvent[] = (events: FaultEvent[], ...)`

**Comment by LaChope** [comment]
@blcham @Kasmadei what is the state of #281 ? If it is solved, I will merge this PR after resolving the conversation.

**Comment by Kasmadei** [comment]
@LaChope 
Hi, i think you can merge. Task is almost done (I have a few changes to merge locally), main part was merged.

**Comment by LaChope** [comment]
@blcham After merge, the UI does not really look great anymore (see screenshot below), should I still merge and create new ticket for it or should I continue here?

image

File: src/components/dialog/faultEvent/FaultEventCreation.tsx
Comment (LaChope) [reply src/components/dialog/faultEvent/FaultEventCreation.tsx:90]: this comes from `const eventTypeWatch = watch("eventType");` so to me it makes sense to leave it like this. Otherwise simply `eventType` would be enough.
(I realize I merged before resolving this, I will make the changes in next PR and reference the commit here if need be)

Comment (blcham) [reply src/components/dialog/faultEvent/FaultEventCreation.tsx:90]: Good point, thanks.

File: src/utils/utils.tsx
Comment (blcham) [reply src/utils/utils.tsx:110]: (events: FaultEvent[], substring: string, newValue: string)

---> 

(events: FaultEvent[], substring: string, newValue: string): FaultEvent[]

Comment (blcham) [reply src/utils/utils.tsx:110]: @LaChope

File: src/components/dialog/faultEvent/FaultEventCreation.tsx
Comment (blcham) [reply src/components/dialog/faultEvent/FaultEventCreation.tsx:196]: @LaChope

Comment (blcham) [reply src/components/dialog/faultEvent/FaultEventCreation.tsx:196]: I mean removing ```gateTypeWatch === GateType.PRIORITY_AND &&```

Comment (LaChope) [reply src/components/dialog/faultEvent/FaultEventCreation.tsx:196]: Resolved with https://github.com/kbss-cvut/fta-fmea-ui/pull/419

File: src/utils/utils.tsx
Comment (blcham) [reply src/utils/utils.tsx:110]: @LaChope

**Pull request merged**
